task8/src/main/java/PostCollectionServlet.java

```diff
@@ -0,0 +1,101 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@WebServlet("/posts/search")
+public class PostCollectionServlet extends HttpServlet {
+    private PostCollection posts;
+    public static final String CONTENT_TYPE = "application/json";
+    public static final String CHARACTER_ENCODING = "UTF-8";
+    public static final String RESULT_NOT_FOUND = "Not found\n";
+    private static final String KEY_AUTHOR = "author";
```

Comment:
Reuse constants; they should not repeat in multiple classes.
Most ofter, in pro projects a dedicated class is created, that would contain only constants. Sort of a "dictionary"